kornia/morphology/morphology.py

```diff
@@ -563,3 +563,55 @@ def bottom_hat(
         )
         - tensor
     )
+
+
+def skeletonize(tensor: torch.Tensor, kernel: Optional[torch.Tensor] = None) -> torch.Tensor:
+    r"""Return the skeleton of a binary image. In morphological terms, it involves reducing foreground regions in a
+    binary image while preserving the extent and connectivity of the original region and throwing away most of the
+    foreground pixels.
+
+    The kernel must have 2 dimensions.
+
+    Args:
+        tensor: Image with shape :math:`(B, 1, H, W)`. [C = 1]
```

Comment:
can we add some logic to accept (B,H,W) and (H,W) /cc @johnnv1

Reply:
makes sense to me -- but this shouldn't work for C>1? since the opening and erode works

Reply:
it makes sense that it wouldn't work. When I commented yesterday, I didn't think of C as being a dimension of channels, but as N masks for a given case (batch)